recursive_art.py

```diff
@@ -1,171 +1,201 @@
-                            random.randint(0, 255))  # Blue channel
-
-    im.save(filename)
-
-
-def generate_art(filename, x_size=350, y_size=350):
-    """Generate computational art and save as an image file.
-
-    Args:
-        filename: string filename for image (should be .png)
-        x_size, y_size: optional args to set image dimensions (default: 350)
-    """
-    # Functions for red, green, and blue channels - where the magic happens!
-    red_function = ["x"]
-    green_function = ["y"]
-    blue_function = ["x"]
-
-    # Create image and loop over all pixels
-    im = Image.new("RGB", (x_size, y_size))
-    pixels = im.load()
-    for i in range(x_size):
-        for j in range(y_size):
-            x = remap_interval(i, 0, x_size, -1, 1)
-            y = remap_interval(j, 0, y_size, -1, 1)
-            pixels[i, j] = (
-                color_map(evaluate_random_function(red_function, x, y)),
-                color_map(evaluate_random_function(green_function, x, y)),
-                color_map(evaluate_random_function(blue_function, x, y))
-            )
-
-    im.save(filename)
-
-
-if __name__ == '__main__':
-    import doctest
-    doctest.testmod()
-
-    # Create some computational art!
-    # TODO: Un-comment the generate_art function call after you
-    #       implement remap_interval and evaluate_random_function
-    # generate_art("myart.png")
-
-    # Test that PIL is installed correctly
-    # TODO: Comment or remove this function call after testing PIL install
-    test_image("noise.png")
+"""TODO: Put your header comment here."""
+
+import random
+import math
+from PIL import Image
+
+
+def build_random_function(min_depth, max_depth):
+    """Build a random function.
+
+    Builds a random function of depth at least min_depth and depth at most
+    max_depth. (See the assignment write-up for the definition of depth
+    in this context)
+
+    Args:
+        min_depth: the minimum depth of the random function
+        max_depth: the maximum depth of the random function
+
+    Returns:
+        The randomly generated function represented as a nested list.
+        (See the assignment writ-eup for details on the representation of
+        these functions)
+    """
+
+    if max_depth > 1:
+        func = random.choice([1, 2, 3, 4])
+        if func == 1:
+            return ["prod", build_random_function(min_depth - 1, max_depth - 1), build_random_function(min_depth - 1, max_depth - 1)]
+        if func==2:
+            return ["cos_pi", build_random_function(min_depth - 1, max_depth - 1)]
+        if func == 3:
+            return ["sin_pi", build_random_function(min_depth - 1, max_depth - 1)]
+        if func == 4:
+            return ["avg", build_random_function(min_depth - 1, max_depth - 1), build_random_function(min_depth - 1, max_depth - 1)]
+    elif max_depth == 1: #Basecase
+        x = ["x"]
+        y = ["y"]
+        return random.choice([x, y])
+    return random.choice([prod, cos, sin, avg])

```

Comment:
Unless `max_depth` is less than or equal to 0, I think this code `return random.choice([prod, cos, sin, avg])` is unreachable, and might not be necessary